src/python/pants/backend/python/util_rules/pex.py

```diff
@@ -1010,5 +1013,48 @@ async def setup_venv_pex_process(request: VenvPexProcess) -> Process:
     )
 
 
+@dataclass(frozen=True)
+class PexDistributionInfo:
+    """Information about an individual distribution in a PEX file, as reported by
+    `PEX_TOOLS=1 repository info -v`."""
+
+    project_name: str
+    version: str
```

Comment:
I guess the consumer could parse what they need, but more out of the box useful useful would be:
```python
version: packaging.version.Version
requires_python: packaging.specifiers.SpecifierSet | None
requires_dists: list[pkg_resources.Requirement]
```